internal/renderer/yatee/driver_test.go

```diff
@@ -0,0 +1,64 @@
+// +build experimental
+
+package yatee
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
+)
+
+var (
+	compose = `version: "3.4"
+services:
+  "@if $myapp.enable":
+    enabledservice:
+      image: alpine:$myapp.alpine_version
+      command: top $$foo
+  other:
+    image: nginx`
+	settings = map[string]interface{}{
+		"myapp": map[string]interface{}{
+			"enable":         true,
+			"alpine_version": "3.7",
+		},
+	}
+	expectedCompose = `services:
+  enabledservice:
+    command: top $$foo
+    image: alpine:3.7
+  other:
+    image: nginx
+version: "3.4"
+`
+)
+
+func TestDriverErrors(t *testing.T) {
+	testCases := []struct {
+		name          string
+		template      string
+		settings      map[string]interface{}
+		expectedError string
+	}{
+		{
+			name:          "no settings",
+			template:      "service: $loop",
+			expectedError: "variable 'loop' not set",
+		},
+	}
+	d := &Driver{}
+	for _, tc := range testCases {
+		_, err := d.Apply(tc.template, tc.settings)
+		assert.Check(t, err != nil)
+		assert.Check(t, is.ErrorContains(err, tc.expectedError))
+	}
+}
+
+func TestDriver(t *testing.T) {
+	d := &Driver{}
```

Comment:
I don't think this needs to be a pointer:
```
  d := Driver{}
```
(or var)

Reply:
the `function` is on a _pointer_ receiver, so yeah, it needs to be a pointer :stuck_out_tongue:

Reply:
Not that simple, see https://github.com/golang/go/wiki/MethodSets#variables